### PR TITLE
test: enable sim test, race test

### PR DESCRIPTION
**Pull request by egonspace**
## Description

- Deleted comment from race test of `test.yml` github action
- Enabled sim test
  - I fixed some bug from the sim test failures
  - don't `GetConsensusParams()` while initializing `checkState`. Reverts a part of https://github.com/line/lbm-sdk/pull/142.
  - use random valid sig block period over 100 for sim testing
  - modify `BaseAccount` marshal/unmarshal
  - add `BaseVestingAccount`, `ContinuousVestingAccount`, `PeriodicVestingAccount` and `DelayedVestingAccount` marshalJSONPB/unmarshalJSONPB
  - `x/distribution/keeper/hooks.go` address conversion bug
  - `x/distribution/types/keys.go` key prefix bug
  - remove param cache: this is a bug. Several `Set`s and no `Write` make the mismatch between db store and cache. Reverts https://github.com/line/lbm-sdk/pull/198
  - `x/slashing/keeper/hooks.go` address conversion bug
- remove invalid keyworkd `SUFFIX_FILTER` from `sims.yml` 
- fix markdown check failures from `docs/core/proto-docs.md`
 
closes: https://github.com/line/lbm-sdk/issues/318

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by iproudhon — APPROVED** [review]

**Comment by iproudhon** [comment]
Looking good to me, other than lint failure.

File: baseapp/baseapp.go [hunk 1]
```diff
@@ -361,7 +361,7 @@ func (app *BaseApp) setCheckState(header ocproto.Header) {
 
 	app.checkState = &state{
 		ms:  ms,
-		ctx: ctx.WithConsensusParams(app.GetConsensusParams(ctx)),
+		ctx: ctx,
```
Comment (egonspace) [inline baseapp/baseapp.go:364]: This modification reverts a part of https://github.com/line/lbm-sdk/pull/142.
The previous code causes sim test failure.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -65,28 +65,28 @@ jobs:
...
-  #         PREFIX_FILTER: |
-  #           cosmovisor
-  #         PATTERNS: |
-  #           **/**.go
-  #           go.mod
-  #           go.sum
-  #     - name: Run cosmovisor tests
-  #       run: cd cosmovisor; make
-  #       if: env.GIT_DIFF
+# TODO: disable test-cosmovisor; this test uses uploaded binary(cosmos-sdk)
```
Comment (egonspace) [inline .github/workflows/test.yml:68]: In order to enable this, research on `test-cosmovisor` is needed. We have to do this as a separate task.

File: .github/workflows/sims.yml [hunk 3]
```diff
@@ -142,8 +141,9 @@ jobs:
         run: go version
       - uses: technote-space/get-diff-action@v5.0.1
         with:
-          SUFFIX_FILTER: |
+          PATTERNSR: |
```
Comment (egonspace) [inline .github/workflows/sims.yml:144]: I found a typo. I'll fix it in next PR or next commit for a comment.

File: x/auth/types/account.go [hunk 4]
```diff
@@ -408,9 +412,10 @@ type GenesisAccount interface {
 // custom json marshaler for BaseAccount & ModuleAccount
 
 type BaseAccountJSON struct {
-	Address  string          `json:"address"`
-	PubKey   json.RawMessage `json:"pub_key"`
-	Sequence string          `json:"sequence"`
+	Address  string `json:"address"`
+	PubType  byte   `json:"pub_type"`
+	PubKey   []byte `json:"pub_key"`
```
Comment (whylee259) [inline x/auth/types/account.go:417]: How about grouping the pubkey properties? 

e.g.
```
{
  "address": ...,
  "sequence": ...,
  "pub_key": {
    "type":...,
    "key":...,
  }
}
```

**Review by whylee259 — APPROVED** [review]
LGTM. I've left a minor comment.

File: x/auth/types/account.go (hunk 4 above)
Comment (egonspace) [reply x/auth/types/account.go:417]: Okay, I apply it.

**Pull request merged**